ArcGISRuntimeSDKQt_CppSamples/EditData/ContingentValues/ContingentValues.cpp

```diff
@@ -0,0 +1,297 @@
+// [WriteFile Name=ContingentValues, Category=EditData]
+// [Legal]
+// Copyright 2022 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "ContingentValues.h"
+
+#include "ArcGISVectorTiledLayer.h"
+#include "CodedValueDomain.h"
+#include "ContingentCodedValue.h"
+#include "ContingentRangeValue.h"
+#include "ContingentValuesDefinition.h"
+#include "ContingentValuesResult.h"
+#include "FeatureLayer.h"
+#include "Geodatabase.h"
+#include "GeodatabaseFeatureTable.h"
+#include "GeometryEngine.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "SimpleFillSymbol.h"
+#include "SimpleLineSymbol.h"
+#include "SimpleRenderer.h"
+
+#include <QDir>
+#include <QtCore/qglobal.h>
+
+#ifdef Q_OS_IOS
+#include <QStandardPaths>
+#endif // Q_OS_IOS
+
+using namespace Esri::ArcGISRuntime;
+
+// helper method to get cross platform data path
+namespace
+{
+QString defaultDataPath()
+{
+  QString dataPath;
+
+#ifdef Q_OS_ANDROID
+  dataPath = "/sdcard";
+#elif defined Q_OS_IOS
+  dataPath = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+#else
+  dataPath = QDir::homePath();
+#endif
+
+  return dataPath;
+}
+}
+
+ContingentValues::ContingentValues(QObject* parent /* = nullptr */):
+  QObject(parent)
+{
+  // Load the basemap from a vector tile package
+  const QString vtpkDataPath = defaultDataPath() + "/ArcGIS/Runtime/Data/vtpk/FillmoreTopographicMap.vtpk";
+  ArcGISVectorTiledLayer* fillmoreVTPK = new ArcGISVectorTiledLayer(QUrl::fromLocalFile(vtpkDataPath), this);
+  Basemap* fillmoreBasemap = new Basemap(fillmoreVTPK, this);
+  m_map = new Map(fillmoreBasemap, this);
+
+  // Load the geodatabase from a mobile geodatabase
+  const QString gdbDataPath = defaultDataPath() + "/ArcGIS/Runtime/Data/geodatabase/ContingentValuesBirdNests.geodatabase";
+  m_geodatabase = new Geodatabase(gdbDataPath, this);
+
+  // The coded value domains in this sample are hardcoded for simplicity, but can be retrieved from the GeodatabaseFeatureTable's Field's Domains.
+  m_statusValues = QVariantMap{{"Occupied", "OCCUPIED"}, {"Unoccupied", "UNOCCUPIED"}};
+  m_protectionValues = QVariantMap{{"Endangered", "ENDANGERED"}, {"Not endangered", "NOT_ENDANGERED"}, {"N/A", "NA"}};
```

Comment:
Did you and Vivian come into an agreement about making this dynamic or keeping it hard coded?